straxen/plugins/event_processing.py

```diff
@@ -733,58 +734,137 @@ def cs2_to_e(self, x):
 
 
 @export
+@strax.takes_config(
+    strax.Option(name='pre_s1_area_threshold', default=1e3,
+                 help='Only take S1s larger than this into account '
+                      'when calculating PeakShadow [PE]'),
+    strax.Option(name='pre_s2_area_threshold', default=1e4,
+                 help='Only take S2s larger than this into account '
+                      'when calculating PeakShadow [PE]'),
+    strax.Option(name='deltatime_exponent', default=-1.0,
+                 help='The exponent of delta t when calculating shadow'),
+    strax.Option(name='time_window_backward', default=int(1e9),
+                 help='Search for peaks casting shadow in this time window [ns]')
+)
 class EventShadow(strax.Plugin):
     """
-    This plugin can calculate shadow at event level.
-    It depends on peak-level shadow.
-    The event-level shadow is its first S2 peak's shadow.
-    If no S2 peaks, the event shadow will be nan.
-    It also gives the position infomation of the previous S2s
-    and main peaks' shadow.
+    This plugin can calculate shadow for main S1 and main S2 in events.
+    It also gives the position information of the previous peaks.
+    References:
+        * v0.1.2 reference: xenon:xenonnt:ac:prediction:shadow_ambience
     """
-    __version__ = '0.0.8'
+    __version__ = '0.1.2'
     depends_on = ('event_basics', 'peak_basics', 'peak_shadow')
     provides = 'event_shadow'
+    data_kind = 'events'
     save_when = strax.SaveWhen.EXPLICIT
 
     def infer_dtype(self):
```

Comment:
I still think that naming is bit confusing. What about naming like this?

`xx_yy_zz_ww`

xx -> variable (dt, ts, ps, pdf, x, y)
yy -> ms1 or ms2
zz -> ps1, ps2, ns1, or ns2
ww -> ts, ps

, where 

ms1/2 -> main s1/2
ps1/2 -> previous s1/2
ts -> largest time_shadow 
ps -> largest position_shadow 
dt -> time_difference 


Then, my proposal is like this:

- s1_pre_dt_s1_time -> dt_ms1_ps1_ts
- s1_pre_dt_s2_time -> dt_ms1_ps2_ts
- s2_pre_dt_s1_time -> dt_ms2_ps1_ts
- s2_pre_dt_s2_time -> dt_ms2_ps2_ts

- s1_pre_x_s2_time -> x_ms1_ps2_ts
- s1_pre_y_s2_time -> y_ms1_ps2_ts
- s2_pre_x_s2_time -> x_ms2_ps2_ts
- s2_pre_y_s2_time -> y_ms2_ps2_ts

- s1_nearest_dt_s1 -> dt_ms1_ns1
- s1_nearest_dt_s2 -> dt_ms1_ns2
- s2_nearest_dt_s1 -> dt_ms2_ns1
- s2_nearest_dt_s2 -> dt_ms2_ns2

- s1_shadow_s1_time -> ts_ms1_ps1_ts
- s1_shadow_s2_time-> ts_ms1_ps2_ts
- s2_shadow_s1_time -> ts_ms2_ps1_ts
- s2_shadow_s2_time-> ts_ms2_ps2_ts

- s1_pre_dt_s2_position -> dt_ms1_ps2_ps
- s2_pre_dt_s2_position -> dt_ms2_ps2_ps

- s1_pre_x_s2_position -> x_ms1_ps2_ps
- s1_pre_y_s2_position -> y_ms1_ps2_ps

- s1_shadow_s2_position -> ps_ms1_ps2_ps
- s2_shadow_s2_position -> ps_ms2_ps2_ps

- s2_pre_x_s2_position -> x_ms2_ps2_ps
- s2_pre_y_s2_position -> y_ms2_ps2_ps

- s1_shadow_s2_position_pdf -> pdf_ms1_ps2_ps (not sure if this is proper naming)
- s2_shadow_s2_position_pdf -> pdf_ms2_ps2_ps (not sure if this is proper naming)

Reply:
@WenzDaniel do you have any suggestions?

Reply:
The purpose to set the names like this is to stay consistent with other names in straxen, for example in `event_info`([doc](https://straxen.readthedocs.io/en/latest/reference/datastructure_nT.html#event-info)). Name of the main S1's features begin with `s1_`, and the name of the main S2's features begin with `s2_`. I keep using `s1_time` & `s2_time` & `s2_position` as suffix because I would call them S1 time shadow, S2 time shadow, S2 position shadow.

Reply:
Maybe rename `s1_shadow_s2_position_pdf` to `s1_pdf_s2_position` would be better.

Reply:
ok, but for example `s1_pre_dt_s1_time` is really confusing to me because `pre_` and `s1_` are separated, while it should be `pre_s1`, so at least something like `s1_dt_ps1_ts` makes sense to me.

Reply:
From my personal point of view I would also prefer a more explicit naming like `s1_dt_preS1_time_shadow`. It adds to readability of the code and I mean during analysis you can always press `tab` to get naming suggestions and select from that.

Reply:
I updated the features' names following reviewers' suggestions. :)